### PR TITLE
Create root directory automatically for download_from_url if not exists

### PR DESCRIPTION
**Pull request by z-Runmin**
Address https://github.com/pytorch/text/issues/760

### CODE REVIEW

File: torchtext/utils.py [hunk 1]
```diff
@@ -78,9 +78,12 @@ def _process_response(r, root, filename):
         root, filename = os.path.split(path)
 
     if not os.path.exists(root):
-        raise RuntimeError(
-            "Download directory {} does not exist. "
-            "Did you create it?".format(root))
+        if root == ".data":
```
Comment (zhangguanheng66) [inline torchtext/utils.py:81]: should we use `try`/`except`? And `root` could be a user-defined directory.

Comment (z-Runmin) [reply torchtext/utils.py]: Sure. Update in 091e263

File: torchtext/utils.py [hunk 2]
```diff
@@ -78,12 +78,11 @@ def _process_response(r, root, filename):
         root, filename = os.path.split(path)
 
     if not os.path.exists(root):
-        if root == ".data":
-            os.makedirs(".data")
-        else:
+        try:
+            os.makedirs(root)
+        except:
```
Comment (zhangguanheng66) [inline torchtext/utils.py:83]: To be save, `except` comes with some error message. I guess you can try `except OSError:` but please double check.

Comment (z-Runmin) [reply torchtext/utils.py]: Yes, I switched to the default `OSError`

**Pull request merged**